src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyout.xaml.cs

```diff
@@ -68,11 +73,10 @@ internal partial class RenameFlyout : InlineRenameAdornment
                 MainPanel.Children.Insert(index + 1, smartRenameControl);
             }
 
-            if (themeService is not null)
-            {
-                Outline.BorderBrush = new SolidColorBrush(themeService.GetThemeColor(EnvironmentColors.AccentBorderColorKey));
-                Background = new SolidColorBrush(themeService.GetThemeColor(EnvironmentColors.ToolWindowBackgroundColorKey));
-            }
+            RefreshColors();
+
+            _editorFormatMap = editorFormatMapService.GetEditorFormatMap("text");
+            _editorFormatMap.FormatMappingChanged += FormatMappingChanged;
```

Comment:
when do we disconnect this event handler though?

Reply:
When `Dispose()` is called. (line 169)

Reply:
sorry.  totally missed that :)